crates/database/src/settlements.rs

```diff
@@ -18,6 +18,25 @@ WHERE
         .await
 }
 
+pub async fn get_hash_by_event(
```

Comment:
I find the naming a bit awkward here because there is no "event" being passed. If it were actually an Event we and we unpacked the (block_number, log_index) inside the method I wouldn't have an issue. 

Also, I think this should return and Optional Hash (because for some block_numbers there may not be a SettlementEvent). It is pretty clear this function is meant to be for a SettlementEvent (since the raw sql is querying the settlement table).

Reply:
As far as I see, `settlements` table is always fully populated, so no need to query optional by hash.
There is also no `by_block_number` query here.
I fixed the event comment.

Reply:
Cool! If your passing a struct that is known to correspond to an entry in the table and not a random pair of integers then the return value does not need to be optional.